erica_app/erica/static/tax_offices.json

```diff
@@ -1 +1 @@
nanzamt Freiberg", "bufa_nr": "3220"}, {"name": "Finanzamt Mittweida", "bufa_nr": "3222"}, {"name": "Finanzamt Plauen", "bufa_nr": "3223"}, {"name": "Finanzamt Stollberg", "bufa_nr": "3224"}, {"name": "Finanzamt Zwickau", "bufa_nr": "3227"}, {"name": "Finanzamt Zschopau", "bufa_nr": "3228"}, {"name": "Finanzamt Leipzig II", "bufa_nr": "3231"}, {"name": "Finanzamt Leipzig I", "bufa_nr": "3232"}, {"name": "Finanzamt Döbeln", "bufa_nr": "3236"}, {"name": "Finanzamt Eilenburg", "bufa_nr": "3237"}, {"name": "Finanzamt Grimma", "bufa_nr": "3238"}, {"name": "Finanzamt Oschatz", "bufa_nr": "3239"}]}, {"state_abbrevation": "st", "name": "Sachsen-Anhalt", "tax_offices": [{"name": "Finanzamt Magdeburg", "bufa_nr": "3102"}, {"name": "Finanzamt Genthin", "bufa_nr": "3103"}, {"name": "Finanzamt Haldensleben", "bufa_nr": "3105"}, {"name": "Finanzamt Salzwedel", "bufa_nr": "3106"}, {"name": "Finanzamt Staßfurt", "bufa_nr": "3107"}, {"name": "Finanzamt Stendal", "bufa_nr": "3108"}, {"name": "Finanzamt Halle (Saale)", "bufa_nr": "3110"}, {"name": "Finanzamt Merseburg", "bufa_nr": "3112"}, {"name": "Finanzamt Dessau-Roßlau", "bufa_nr": "3114"}, {"name": "Finanzamt Wittenberg", "bufa_nr": "3115"}, {"name": "Finanzamt Bitterfeld-Wolfen", "bufa_nr": "3116"}, {"name": "Finanzamt Quedlinburg", "bufa_nr": "3117"}, {"name": "Finanzamt Eisleben", "bufa_nr": "3118"}, {"name": "Finanzamt Naumburg", "bufa_nr": "3119"}]}, {"state_abbrevation": "sh", "name": "Schleswig-Holstein", "tax_offices": [{"name": "Finanzamt Bad Segeberg", "bufa_nr": "2111"}, {"name": "Finanzamt Elmshorn", "bufa_nr": "2113"}, {"name": "Finanzamt Flensburg", "bufa_nr": "2115"}, {"name": "Finanzamt Dithmarschen", "bufa_nr": "2116"}, {"name": "Finanzamt Nordfriesland", "bufa_nr": "2117"}, {"name": "Finanzamt Itzehoe", "bufa_nr": "2118"}, {"name": "Finanzamt Kiel", "bufa_nr": "2120"}, {"name": "Finanzamt Lübeck", "bufa_nr": "2122"}, {"name": "Finanzamt Neumünster", "bufa_nr": "2124"}, {"name": "Finanzamt Ostholstein", "bufa_nr": "2125"}, {"name": "Finanzamt Plön", "bufa_nr": "2126"}, {"name": "Finanzamt Ratzeburg", "bufa_nr": "2127"}, {"name": "Finanzamt Rendsburg", "bufa_nr": "2128"}, {"name": "Finanzamt Eckernförde-Schleswig", "bufa_nr": "2129"}, {"name": "Finanzamt Stormarn", "bufa_nr": "2130"}, {"name": "Finanzamt Pinneberg", "bufa_nr": "2131"}]}, {"state_abbrevation": "th", "name": "Thüringen", "tax_offices": [{"name": "Finanzamt Erfurt", "bufa_nr": "4151"}, {"name": "Finanzamt Ilmenau", "bufa_nr": "4154"}, {"name": "Finanzamt Eisenach", "bufa_nr": "4155"}, {"name": "Finanzamt Gotha", "bufa_nr": "4156"}, {"name": "Finanzamt Mühlhausen", "bufa_nr": "4157"}, {"name": "Finanzamt Sondershausen", "bufa_nr": "4159"}, {"name": "Finanzamt Gera", "bufa_nr": "4161"}, {"name": "Finanzamt Jena", "bufa_nr": "4162"}, {"name": "Finanzamt Pößneck", "bufa_nr": "4165"}, {"name": "Finanzamt Altenburg", "bufa_nr": "4166"}, {"name": "Finanzamt Sonneberg", "bufa_nr": "4170"}, {"name": "Finanzamt Suhl", "bufa_nr": "4171"}]}]}
```

Comment:
Wait, I'm confused… The json file _did_ have correct unicode escapes? Am I misremembering or did your browser show mangled characters when you demoed it yesterday?

IMO unicode escapes (`\u....`) would be generally preferable to encoding values in UTF-8 because it's maximally compatible that way.

Reply:
It did show mangled characters. I've changed the script to use UTF-8 encoding when generating the file: 

```python
 with open(_TAX_OFFICES_JSON_FILE_NAME, 'w', encoding="utf-8") as tax_offices_file:
         tax_offices_file.write(json.dumps(tax_office_list))
```

As we then also send the file with `media_type="application/json; charset=utf-8"` this will not show mangled characters. We could also store the file different. However, I think this makes it easy to send the file in the correct formatting, don't you think?

Reply:
Hm, I'm not so keen on jumping through hoops to create a utf-8 JSON files when an ascii version with unicode escapes is arguably better. It seems to me that the previous file was perfectly fine, and that the mangled characters must have had another source?

Don't take my word, try it out (ba913a9127cbc49b41ef2d3020ad2b0173706f60 is the previous version):
```
$ python3
>>> import requests
>>> requests.get('https://raw.githubusercontent.com/digitalservice4germany/steuerlotse/ba913a9127cbc49b41ef2d3020ad2b0173706f60/erica_app/erica/static/tax_offices.json').json()['tax_offices'][0]
{'county_abbrevation': 'bw', 'name': 'Baden-Württemberg', 'tax_offices': [{'name': 'Finanzamt Offenburg Außenstelle Achern', 'bufa_nr': '2801'}, {'name': 'Finanzamt Villingen-Schwenningen Außenstelle Donaueschingen', ...
```

It loads everything as it should. Same thing if I put it in an online tool like https://jsonformatter.org/#.

So I think we should revert to the previous way of generating the file and figure out where the mangled characters got created. (Btw, I believe we shouldn't have to set a `media_type` on the FileResponse as long as the file ends in `.json`: ["media_type - A string giving the media type. If unset, the filename or path will be used to infer a media type."](https://www.starlette.io/responses/#fileresponse)

Reply:
> It seems to me that the previous file was perfectly fine, and that the mangled characters must have had another source?

I think the source for the mangled characters is the browser because they are also shown if I request `https://raw.githubusercontent.com/digitalservice4germany/steuerlotse/ba913a9127cbc49b41ef2d3020ad2b0173706f60/erica_app/erica/static/tax_offices.json`directly. Firefox will show it correctly because it parses JSON.
`UTF-8` is the default/only encoding for JSON ([s. here](https://datatracker.ietf.org/doc/html/rfc8259#section-8.1)). Therefore, whenever we parse it, it is correctly decoded (in python, as well as Firefox and with the jsonformatter tool). You can see that also below. Safari and Chrome do not seem to be parsing it as JSON but rather just showing it like a text file -> mangled characters.

Basically, we can use the old file, get rid of all the utf-8 encoding loops and it works as before (Browser will show the mangled characters but webapp won't).

```python
$ python3
>>> r = requests.get('https://raw.githubusercontent.com/digitalservice4germany/steuerlotse/ba913a9127cbc49b41ef2d3020ad2b0173706f60/erica_app/erica/static/tax_offices.json')
>>>r.encoding
'utf-8'
```

> Btw, I believe we shouldn't have to set a media_type on the FileResponse as long as the file ends in .json: "media_type - A string giving the media type. If unset, the filename or path will be used to infer a media type."

Yes, you are correct. I've only had to add that because of the browser.